Frameworks/Foundation/NSScanner.mm

```diff
@@ -25,547 +25,311 @@
-*/
-+ (instancetype)scannerWithString:(id)str {
-    NSScanner* ret = [self alloc];
+    if (!scanEnd || scanEnd == scanStart) {
+        return NO;
+    }
 
-    return [[ret initWithString:str] autorelease];
+    if (valuep) {
+        *valuep = value;
+    }
+
+    self.scanLocation = start + (scanEnd - scanStart);
+    return YES;
 }
 
-/**
- @Status Interoperable
-*/
-- (instancetype)initWithString:(id)string {
-    self = [self init];
-    _string = [string copy];
-    _location = 0;
-    _len = [_string length];
-    _skipSet = [[NSCharacterSet whitespaceAndNewlineCharacterSet] retain];
-    _isCaseSensitive = NO;
-    _locale = nil;
+// Overload of __scanNumeric for CRT functions that do not take a base parameter
+template <typename TNumeric>
+static BOOL __scanNumeric(NSScanner* self,
+                          TNumeric* valuep,
+                          TNumeric (*localeFunc)(const wchar_t*, wchar_t**, _locale_t), // eg: _wcstod_l
+                          TNumeric (*nonLocaleFunc)(const wchar_t*, wchar_t**), // eg: wcstod
+                          bool enforceUnsigned = false) {
+    // Skip past the characters in charactersToBeSkipped
+    NSUInteger start = [self _indexOfNextUnskippedCharacter];
+    if (start >= self->_stringChars.size()) {
+        return NO;
+    }
 
-    return self;
-}
+    wchar_t* scanStart = reinterpret_cast<wchar_t*>(self->_stringChars.data() + start);
+    wchar_t* scanEnd;
 
-/**
- @Status Interoperable
-*/
-- (NSString*)string {
-    return _string;
+    // Early return for certain characters not in charactersToBeSkipped
+    // * Whitespace can be removed from charactersToBeSkipped but still skipped by a CRT function
+    // * '-' is skipped without failing in unsigned CRT functions
+    if (iswspace(*scanStart) || ((enforceUnsigned) && (*scanStart == '-'))) {
+        return NO;
+    }
+
+    TNumeric value = self->_crtLocale ? localeFunc(scanStart, &scanEnd, self->_crtLocale) : nonLocaleFunc(scanStart, &scanEnd);
+
+    if (!scanEnd || scanEnd == scanStart) {
+        return NO;
+    }
+
+    if (valuep) {
+        *valuep = value;
+    }
+
+    self.scanLocation = start + (scanEnd - scanStart);
+    return YES;
 }
 
-/**
- @Status Interoperable
-*/
-- (void)dealloc {
-    [_string release];
-    [_skipSet release];
-    [_locale release];
-    [super dealloc];
+@implementation NSScanner {
+@public
+    std::vector<unichar> _stringChars;
+
+    StrongId<NSLocale> _locale;
+    _locale_t _crtLocale;
 }
 
 /**
  @Status Interoperable
 */
-- (NSCharacterSet*)charactersToBeSkipped {
-    return _skipSet;
++ (instancetype)scannerWithString:(NSString* _Nonnull)string {
+    return [[[self alloc] initWithString:string] autorelease];
 }
 
 /**
  @Status Interoperable
 */
-- (BOOL)caseSensitive {
-    return _isCaseSensitive;
++ (instancetype)localizedScannerWithString:(NSString* _Nonnull)string {
+    NSScanner* ret = [self scannerWithString:string];
+    ret.locale = [NSLocale currentLocale];
```

Comment:
there is no equivalent public init, seems like unnecessary frills to make one just for this